BDArmory/Modules/RocketLauncher.cs

```diff
@@ -437,7 +453,23 @@ public override void OnUpdate()
                 }
             }
         }
-
+		void FixedUpdate()
+		{
+			if (HighLogic.LoadedSceneIsFlight && !vessel.packed)
+			{
+				if (vessel.isActiveVessel)
+				{
+					if (rocketsLeft > 0 && ammoGauge == null) //only redraw these if nulled from vessel switch
+					{
+						UpdateAmmoMeter();
+					}
+					if (rocketsLeft <= 0 && emptyGauge == null)
+					{
+						UpdateEmptyAlert();
+					}
+				}
```

Comment:
This looks like UI stuff to me, shouldn't it go into `Update` instead?

Also, this one is definitely not on you, but it is confusing to have both `FixedUpdate` and `OnFixedUpdate` on the same class (at least to me), might be better to merge them both into `FixedUpdate`. Same applies if it is moved to `Update`.

Reply:
Yes, it should be in Update(); see above.